### PR TITLE
test: Code Coverage: Add Edit Mileage Setup

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 7a9fd9b</samp>

This change refactors the unit tests for the `AddEditMileagePage` component by splitting the original spec file into smaller files and using a dynamic test bed function. This improves the readability, maintainability, and reliability of the tests.

### <samp>🤖 Generated by Copilot at 7a9fd9b</samp>

> _`AddEditMileagePage`_
> _Refactored spec file, tests_
> _Autumn leaves cleanup_

### <samp>🤖 Generated by Copilot at 7a9fd9b</samp>

*  Refactor the unit tests for AddEditMileagePage component into smaller and more manageable files ([link](https://github.com/fylein/fyle-mobile-app/pull/2260/files?diff=unified&w=0#diff-5ea8feba7a8355dcc3a878d59e6ac3c56d89e15472f7ddfe5d9c63c38a90c961R1-R181), [link](https://github.com/fylein/fyle-mobile-app/pull/2260/files?diff=unified&w=0#diff-fc91899235a18a122819ae264aaa52949e9e0f7d6253e57d25adcce9a683c0b5))
  - Delete the original `add-edit-mileage.page.spec.ts` file ([link](https://github.com/fylein/fyle-mobile-app/pull/2260/files?diff=unified&w=0#diff-fc91899235a18a122819ae264aaa52949e9e0f7d6253e57d25adcce9a683c0b5))
  - Create a new file `add-edit-mileage-1.spec.ts` that contains the imports, declarations, and initializations of the dependencies and services required for the component and its tests ([link](https://github.com/fylein/fyle-mobile-app/pull/2260/files?diff=unified&w=0#diff-5ea8feba7a8355dcc3a878d59e6ac3c56d89e15472f7ddfe5d9c63c38a90c961R1-R181))
  - Define a function `TestCases1` that wraps the describe block for the first set of tests, which cover the component's initialization, inputs, outputs, and form controls ([link](https://github.com/fylein/fyle-mobile-app/pull/2260/files?diff=unified&w=0#diff-5ea8feba7a8355dcc3a878d59e6ac3c56d89e15472f7ddfe5d9c63c38a90c961R1-R181))
  - Use a dynamic way of injecting the test bed for each test case, by passing a function that returns the test bed configuration as an argument to `TestCases1` ([link](https://github.com/fylein/fyle-mobile-app/pull/2260/files?diff=unified&w=0#diff-5ea8feba7a8355dcc3a878d59e6ac3c56d89e15472f7ddfe5d9c63c38a90c961R1-R181))

## Clickup
https://app.clickup.com/t/85ztk92ck

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/services/per-diem.service.ts [hunk 1]
```diff
@@ -48,11 +49,11 @@ export class PerDiemService {
           const allowedPerDiemIds = settings.per_diem_rate_settings.allowed_per_diem_ids;
 
           if (allPerDiemRates?.length > 0) {
-            allowedPerDiems = allPerDiemRates.filter((perDiem) => allowedPerDiemIds.includes(perDiem.id));
+            allowedPerDiems = allPerDiemRates.filter((perDiem) => allowedPerDiemIds.includes(perDiem.id as never));
```
Comment (Chethan-Fyle) [inline src/app/core/services/per-diem.service.ts:52]: Why `never`? 🤔

File: src/app/core/services/file.service.ts [hunk 2]
```diff
@@ -88,15 +88,15 @@ export class FileService {
     return file;
   }
 
-  post(fileObj: File | Record<string, string> | FileObject) {
+  post(fileObj: File | Record<string, string> | FileObject): Observable<unknown> {
```
Comment (Chethan-Fyle) [inline src/app/core/services/file.service.ts:91]: Why `unknown`? 🤔

File: src/app/core/services/timezone.service.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { Injectable } from '@angular/core';
 import { cloneDeep } from 'lodash';
-import { UtilityService } from './utility.service';
 import { TxnCustomProperties } from '../models/txn-custom-properties.model';
+import { UtilityService } from './utility.service';
```
Comment (Chethan-Fyle) [inline src/app/core/services/timezone.service.ts:4]: Is this used in this file?

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/core/services/timezone.service.ts (hunk 3 above)
Comment (jayfyle) [reply src/app/core/services/timezone.service.ts]: Yes, one of the methods uses this service

**Pull request merged**